src/app/features/exchange-set/ess-add-single-encs/ess-add-single-encs.component.ts

```diff
@@ -1,14 +1,62 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
+import { Router } from '@angular/router';
+import { EssUploadFileService } from '../../../core/services/ess-upload-file.service';
 
 @Component({
   selector: 'app-ess-add-single-encs',
   templateUrl: './ess-add-single-encs.component.html',
   styleUrls: ['./ess-add-single-encs.component.scss']
 })
 export class EssAddSingleEncsComponent implements OnInit {
-  
-  constructor() { }
+  @Input() renderedFrom: string;
+  @Input() btnText: string;
+
+  @ViewChild('ukhoTarget') ukhoDialog: ElementRef;
+  messageType: 'info' | 'warning' | 'success' | 'error' = 'info';
+  messageDesc = '';
+  displayErrorMessage = false;
+
+  txtSingleEnc: string = "";
+
+  constructor(private essUploadFileService: EssUploadFileService,
+    private route: Router) { }
 
   ngOnInit(): void {
   }
+
+  validateAndAddENC() {
+    if (this.renderedFrom == 'encList') {
+
+    }
+    else if ((this.renderedFrom == 'essHome')) {
+      this.addSingleEncToList();
+    }
+  }
+
+  addSingleEncToList() {
+    if (this.txtSingleEnc != '') {
+      if (this.essUploadFileService.validateENCFormat(this.txtSingleEnc)) {
+        this.displayErrorMessage = false;
+        this.essUploadFileService.setValidSingleEnc(this.txtSingleEnc);
```

Comment:
Please set `essUploadFileService.infoMessage` to false here to prevent previously displayed error/warning messages on the list page being displayed again

Reply:
Resolved. Thanks